helsingborg-stad/app-mitt-helsingborg
# Add spacing between children in summary

## Explain the changes you’ve made

I've added a spacer component for multiple children in a SummaryList, in accordance with #CU-2qa0ng4

## Explain your solution

For multiple items in a summary list, I add a spacer object, which is then used to push a spacer component into the array of rendered children.

## How to test

1. Checkout this branch
2. Start grundansökan
3. Add at least 2 children
4. Go to the first summary
5. See that there's now a space between the two children

## Tested environments

- [] Storybook on a iOS device/simulator.
- [] Building the Application on a Android device/simulator.

## Screenshots

Before:

![image](https://user-images.githubusercontent.com/10225982/169467233-37094eef-1e13-46f4-8ae7-6da68063aab0.png)


After:

![image](https://user-images.githubusercontent.com/10225982/169467264-df10aed3-5f2b-4dcd-a76d-269ebca7384b.png)


## Other notes

---

## Checklist for reviewers

Here are a few points that are useful for reviewers to check.

The short list of the (maybe) most important points are:

- Does this PR work as expected?
- Does the PR properly only concern a single feature/bug/thing?
- Is the code clear and concise?
- Will the changes impact existing functionality?
- Are there any linting errors (ESLint) that should be resolved?
- Do the automatic GitHub checks pass?

However it is good if a reviewer can be as thorough as possible,
within reasonable limits of course.

### Functionality

- Does this PR work as expected?
- If this is a new feature, does it add value, or is it a sign of feature-creep?
- Will the changes impact existing functionality?
- Does the changes create inconsistencies somewhere else?
- Does the code work on different platforms if applicable?
- How does the code handle if unexpected values are passed to functions etc.?

### Readability, syntax, formatting, style

- Is the code clear and concise?
- Has styling rules (Prettier) been applied to included files?
- Are functions/variables/etc. aptly named?

### Design

- Is the code properly planned and designed?
- Is code organized/separated in a logical fashion?
- Does the code follow clean code practices to a reasonable degree?
- Is the code self-documenting enough?
- Is the code in sync with existing patterns/technologies?
- Does the code reuse existing functionality where applicable?
- Is the code designed with reusability in mind?
- Does the code have hard-coded values that can be refactored?

### Patterns, idioms, tests, best practices

- Are there any linting errors (ESLint) that should be resolved?
- Is the code properly typed (TypeScript)?
- Does the code utilize language features to their full potential?

### Debugging, testing, configuring

- Is appropriate logging used?
- Have tests been added/modified to accommodate the changes?
- Is the feature sufficiently configurable in case tweaks are required in the future?

### Performance, reliability, scalability

- Is the code well optimized?
- Does the code scale?
- Does the code handle failure scenarios?

### Security

- Is there any obvious potential for exploits and security threats?
- Does code check proper authorization where applicable?

### Etiquettes

- Are commits atomic (does only one thing per commit)?
- Does the PR properly only concern a single feature/bug/thing?
- Are commit messages well-written?
- Do the automatic GitHub checks pass?
